src/test/java/com/github/pagehelper/dialect/AbstractRowBoundsDialectTest.java

```diff
@@ -0,0 +1,80 @@
+package com.github.pagehelper.dialect;
+
+import com.diffblue.deeptestutils.Reflector;
+import com.github.pagehelper.PageRowBounds;
+import com.github.pagehelper.dialect.rowbounds.Db2RowBoundsDialect;
+import org.apache.ibatis.mapping.MappedStatement;
+import org.apache.ibatis.session.RowBounds;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import java.lang.reflect.InvocationTargetException;
+
+public class AbstractRowBoundsDialectTest {
+
+  @Rule public ExpectedException thrown = ExpectedException.none();
+
+  /* testedClasses: AbstractRowBoundsDialect */
+
+  /*
+   * Test generated by Diffblue Deeptest.
+   * This test case covers:
+   *  - conditional line 50 branch to line 51
+   *  - conditional line 51 branch to line 51
+   *  - conditional line 52 branch to line 52
+   */
+
+  @Test
+  public void beforeCountInputNotNullZeroNotNullOutputFalse() throws InvocationTargetException {
+
+    // Arrange
+    final Db2RowBoundsDialect objectUnderTest = ((Db2RowBoundsDialect)Reflector.getInstance(
+        "com.github.pagehelper.dialect.rowbounds.Db2RowBoundsDialect"));
+    objectUnderTest.countSqlParser = null;
+    final MappedStatement ms =
+        ((MappedStatement)Reflector.getInstance("org.apache.ibatis.mapping.MappedStatement"));
+    final Object parameterObject = 0;
+    final PageRowBounds rowBounds =
+        ((PageRowBounds)Reflector.getInstance("com.github.pagehelper.PageRowBounds"));
+    Reflector.setField(rowBounds, "count", false);
+    Reflector.setField(rowBounds, "total", null);
+
+    // Act
+    final boolean retval = objectUnderTest.beforeCount(ms, parameterObject, rowBounds);
+
+    // Assert result
+    Assert.assertEquals(false, retval);
+  }
+
+  /*
+   * Test generated by Diffblue Deeptest.
+   * This test case covers:
+   *  - conditional line 50 branch to line 51
+   *  - conditional line 51 branch to line 51
+   *  - conditional line 52 branch to line 52
+   */
+
+  @Test
+  public void beforeCountInputNotNullZeroNotNullOutputTrue() throws InvocationTargetException {
```

Comment:
The already have a test that covers this line according to the online demo. Might be worth leaving out.

Reply:
Removed.